### PR TITLE
Created new type definition for chargebee react wrapper

### PR DESCRIPTION
**Pull request by sagar7993**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/chargebee__chargebee-js-react-wrapper/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
+// Type definitions for chargebee-js-wrappers 0.2
+// Project: https://www.npmjs.com/package/@chargebee/chargebee-js-react-wrapper
+// Definitions by: Sagar Jain <https://github.com/sagar7993>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { Component, CSSProperties } from 'react';
+
+export interface TokenizeAdditionalData {
+	readonly firstName?: string;
```
Comment (danger-public) [inline types/chargebee__chargebee-js-react-wrapper/index.d.ts:9]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/chargebee__chargebee-js-react-wrapper/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    readonly billingAddress?: AuthorizeWith3dsAdditionalDataAddress;
+    readonly customerBillingAddress?: AuthorizeWith3dsAdditionalDataAddress;
+    readonly shippingAddress?: AuthorizeWith3dsAdditionalDataAddress;
+}
+export interface PaymentIntent {
+    readonly id: string;
+    readonly status: 'inited' | 'in_progress' | 'authorized' | 'consumed' | 'expired';
+	/**
+	 * The currency code (ISO 4217 format) of the amount used in transaction.
+	 */
```
Comment (DanielRosenwasser) [inline types/chargebee__chargebee-js-react-wrapper/index.d.ts:45]:
```suggestion
    /**
     * The currency code (ISO 4217 format) of the amount used in transaction.
     */
```

File: types/chargebee__chargebee-js-react-wrapper/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    readonly shippingAddress?: AuthorizeWith3dsAdditionalDataAddress;
+}
+export interface PaymentIntent {
+    readonly id: string;
+    readonly status: 'inited' | 'in_progress' | 'authorized' | 'consumed' | 'expired';
+    /**
+     * The currency code (ISO 4217 format) of the amount used in transaction.
+     */
+    readonly currency_code?: string;
+	/**
```
Comment (danger-public) [inline types/chargebee__chargebee-js-react-wrapper/index.d.ts:47]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## babel__core ([<kbd>unpkg</kbd>](https://unpkg.com/browse/babel__core@latest/))
was missing the following properties:
1. Plugin
1. File
1. buildExternalHelpers
1. getEnv
1. OptionManager

<details>
<summary>as well as these 5 other properties...</summary>
<p>Plugin, File, buildExternalHelpers, getEnv, OptionManager</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6675c92a26937491ba991ed9fa5a9f65f6a9f372
</p>

**Comment by elibarzilay** [comment]
@sagar7993: what's the state of this?  Also, are the tslint overrides needed?

**Comment by elibarzilay** [comment]
@sagar7993: re-ping

File: types/chargebee__chargebee-js-react-wrapper/tslint.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-void-expression": false,
```
Comment (sandersn) [inline types/chargebee__chargebee-js-react-wrapper/tslint.json:4]: is this used? I didn't see any void expressions in the new code.

File: types/chargebee__chargebee-js-react-wrapper/tslint.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        "only-arrow-functions": false,
```
Comment (sandersn) [inline types/chargebee__chargebee-js-react-wrapper/tslint.json:5]: same -- I didn't see any function expressions.

**Review by sandersn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**